### ISSUE TITLE
How to evaluate the vram

### ISSUE THREAD

**Issue by shuxiaobo** [jdwx/gptj-finetune#2]
Hello, thank you very much for contributing this repo, it has helped me a lot, but I have a question about vram calculation, the code sets max_len = 2048, and the readme says each sample = 2.6GB, how is this calculated? Also, what is 0.7GB used for?

Thanks for you reply

**Comment by jdwx** [comment]
The max_len is a function of the model; it is the n_ctx value on the model card.

The estimates given were approximated by doing linear regression on batch sizes 1-7 and calculating the slope (per-batch overhead) and intercept (inherent overhead).

The model is opaque and Deepspeed is complex, so I couldn't speculate what the VRAM is being used for besides samples.  Something to do with the gradient accumulation would be my best guess.
